### PR TITLE
feat(CFW): add a  data source to get the list of CFW attack logs

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add a  data source to get the list of CFW attack logs
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add a  data source to get the list of CFW attack logs
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST="./huaweicloud/services/acceptance/cfw" TESTARGS="-run TestAccDataSourceCfwAttackLogs_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cfw -v -run TestAccDataSourceCfwAttackLogs_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceCfwAttackLogs_basic
=== PAUSE TestAccDataSourceCfwAttackLogs_basic
=== CONT  TestAccDataSourceCfwAttackLogs_basic
--- PASS: TestAccDataSourceCfwAttackLogs_basic (166.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cfw       166.416s

```

### CODE REVIEW

File: huaweicloud/services/cfw/data_source_huaweicloud_cfw_attack_logs.go [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+		curJson := utils.PathSearch("data.records", respBody, make([]interface{}, 0))
+		curArray := curJson.([]interface{})
+
+		if len(curArray) == 0 {
+			break
+		}
+		// The event_time and log_id determine the starting position of the next page.
+		nextDate := utils.PathSearch("[-1].event_time", curJson, 0.0).(float64)
+		logID := utils.PathSearch("[-1].log_id", curJson, "")
+		path = fmt.Sprintf("%s&offset=0&next_date=%v&log_id=%v", basePath, int64(nextDate), logID)
```
Comment (houpeng80) [inline huaweicloud/services/cfw/data_source_huaweicloud_cfw_attack_logs.go:325]: the param of offset may be useless

File: huaweicloud/services/cfw/data_source_huaweicloud_cfw_attack_logs.go [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+		return nil, err
+	}
+
+	curJson := utils.PathSearch("data.records", respBody, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	if len(curArray) > 0 {
+		rst = append(rst, curArray...)
+	}
+
+	return rst, nil
```
Comment (houpeng80) [inline huaweicloud/services/cfw/data_source_huaweicloud_cfw_attack_logs.go]:
```suggestion
	
	return curJson.([]interface{}), nil
```

Comment (profoundwu) [reply huaweicloud/services/cfw/data_source_huaweicloud_cfw_attack_logs.go]: fixed.

**Pull request merged**